### PR TITLE
[GSP-13] Set up a JDBC repository for Messages and Users

### PR DESCRIPTION
**Pull request by ktrof**

### CODE REVIEW

File: backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    private User user = new User();
+
+    @Override
+    public boolean createUser(User user) {
+        try (PreparedStatement preparedStatement = DbUtil.getConnection().prepareStatement(addUserSql)) {
+
+            preparedStatement.setString(1, user.getLogin());
+            preparedStatement.setString(2, user.getPassword());
+
+            preparedStatement.executeUpdate();
```
Comment (nyaankato) [inline backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java]: are you trying to execute it twice?

File: backend/src/main/java/com/gridstudentpractice/chatservice/service/DbUserServiceImpl.java [hunk 2]
```diff
@@ -1,70 +1,23 @@
...
-        } catch (SQLException e) {
-            e.printStackTrace();
-        }
-        return true;
+    public void addUser(User user) {
+        userRepository.createUser(user);
     }
 
     @Override
     public User getUser(String loginToGet) {
```
Comment (nyaankato) [inline backend/src/main/java/com/gridstudentpractice/chatservice/service/DbUserServiceImpl.java]: pls rename loginToGet to smth like "userLogin"

File: backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            preparedStatement.close();
+
+        } catch (SQLException e) {
+            throw new RepositoryException("User creation error", e);
+        }
+        return true;
+    }
+
+    @Override
+    public User readUser(String loginToGet) {
```
Comment (nyaankato) [inline backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java]: getUserByLogin(String userLogin) would be better

File: backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+@Repository
+public class JDBCUserRepositoryImpl implements UserRepository {
+
+    final static String addUserSql = "INSERT INTO users (login, password) VALUES (?, ?)" ;
+    final static String checkUserSql = "SELECT * FROM users WHERE login = ?";
```
Comment (nyaankato) [inline backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java]: shouldn't quotes be here?

File: backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCMessageRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class JDBCMessageRepositoryImpl implements MessageRepository {
+
+    private List<Message> messages = new ArrayList<>();
```
Comment (nyaankato) [inline backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCMessageRepositoryImpl.java]: do we need this?

File: backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCMessageRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        String mBody = message.getBody();
+        String mTime = message.getTimestamp().toString();
+
+        try (PreparedStatement preparedStatement = DbUtil.getConnection().prepareStatement(insertTableSql)) {
+
+            preparedStatement.setString(1, mSender);
+            preparedStatement.setString(2, mBody);
+            preparedStatement.setString(3, mTime);
+
+            preparedStatement.executeUpdate();
```
Comment (nyaankato) [inline backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCMessageRepositoryImpl.java]: are you trying to execute it twice? :)

File: backend/src/test/java/com/gridstudentpractice/chatservice/controller/ResponseBodyTests.java [hunk 7]
```diff
@@ -53,35 +53,35 @@ private Message validMessage() {
...
+//                .thenReturn(message);
+//        RequestBuilder requestBuilder = MockMvcRequestBuilders
+//                .post("/restChat")
+//                .accept(MediaType.APPLICATION_JSON)
+//                .content(body)
+//                .contentType(MediaType.APPLICATION_JSON);
+//        MvcResult mvcResult = mvc.perform(requestBuilder).andReturn();
+//        MockHttpServletResponse response = mvcResult.getResponse();
+//        assertEquals(HttpStatus.BAD_REQUEST.value(), response.getStatus());
+//    }
```
Comment (Mantarus) [inline backend/src/test/java/com/gridstudentpractice/chatservice/controller/ResponseBodyTests.java]: Why tests are commented out? They aren't valid anymore? If not, could we update the tests to make them valid again?

File: backend/src/main/java/com/gridstudentpractice/chatservice/service/DbUserServiceImpl.java (hunk 2 above)
Comment (Mantarus) [reply backend/src/main/java/com/gridstudentpractice/chatservice/service/DbUserServiceImpl.java]: Yep, we are getting user here, not login

File: backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java (hunk 3 above)
Comment (Mantarus) [reply backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java]: we can go with just `getUser`. It accepts user login as parameter so there is no need to declare that again in the name of the method :)

File: backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java (hunk 4 above)
Comment (Mantarus) [reply backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java]: which quotes?

File: backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                while (rs.next()) {
+
+                    user.setId(rs.getInt("id"));
+                    user.setLogin(rs.getString("login"));
+                    user.setPassword(rs.getString("password"));
+                }
+
+                preparedStatement.close();
+
+                if (!(user.getLogin().equals("null") || user.getPassword().equals("null")))
```
Comment (Mantarus) [inline backend/src/main/java/com/gridstudentpractice/chatservice/repository/JDBCUserRepositoryImpl.java]: why you are comparing user login and user password with "null" string literal? If you need to compare this with null, you can simply write `user.getLogin() == null`

**Pull request closed without merging**